modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataKeyIdObjectResolver.java

```diff
@@ -262,6 +319,11 @@ public Object getResultId() throws MetadataResolvingException {
   private List<ParameterModel> getMetadataKeyParts(ComponentModel componentModel) {
     return componentModel.getAllParameterModels().stream()
         .filter(p -> p.getModelProperty(MetadataKeyPartModelProperty.class).isPresent())
+        .sorted((p1, p2) -> {
+          Optional<MetadataKeyPartModelProperty> mk1 = p1.getModelProperty(MetadataKeyPartModelProperty.class);
```

Comment:
this would return a sorted list of model properties. What he wants is the sorted list of parameters.